src/requirements/__test__/requirement-graph.test.ts

```diff
@@ -0,0 +1,63 @@
+import RequirementFulfillmentGraph from '../requirement-graph';
+
+it('RequirementFulfillmentGraph works.', () => {
```

Comment:
I think these tests would be sufficient on its own. It always covers some common setup and tricky cases (e.g. removing on edge causing a requirement to have no linked courses). Additional unit tests should be done on business logic itself instead of this abstract graph. e.g. testing whether all double counting has been correctly eliminated in a phase.

Reply:
Yeah that's more what I was thinking about, but you're right the graph itself should be good to go